central/graphql/resolvers/image_vulnerabilities_test.go

```diff
@@ -0,0 +1,364 @@
+	imageCompIndexer := imageComponentPostgres.NewIndexer(s.db)
+	imageCompSearcher := imageComponentSearch.NewV2(imageCompStore, imageCompIndexer)
+	s.resolver.ImageComponentDataStore = imageComponentDataStore.New(nil, imageCompStore, imageCompIndexer, imageCompSearcher, riskMock, ranking.NewRanker())
+
+	// Sac permissions
+	s.ctx = sac.WithAllAccess(s.ctx)
+
+	// loaders used by graphql layer
+	loaders.RegisterTypeFactory(reflect.TypeOf(storage.Image{}), func() interface{} {
+		return loaders.NewImageLoader(s.resolver.ImageDataStore)
+	})
+	loaders.RegisterTypeFactory(reflect.TypeOf(storage.ImageComponent{}), func() interface{} {
+		return loaders.NewComponentLoader(s.resolver.ImageComponentDataStore)
+	})
+	loaders.RegisterTypeFactory(reflect.TypeOf(storage.ImageCVE{}), func() interface{} {
+		return loaders.NewImageCVELoader(s.resolver.ImageCVEDataStore)
+	})
+	s.ctx = loaders.WithLoaderContext(s.ctx)
+
+	// Add Test Data to DataStores
+	testImages := testImages()
+	for _, image := range testImages {
+		err = imageStore.Upsert(s.ctx, image)
+		s.NoError(err)
+	}
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TearDownSuite() {
+	imagePostgres.Destroy(s.ctx, s.db)
+	imageComponentPostgres.Destroy(s.ctx, s.db)
+	imageCVEPostgres.Destroy(s.ctx, s.db)
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestUnauthorizedImageVulnerabilityEndpoint() {
+	_, err := s.resolver.ImageVulnerability(s.ctx, IDQuery{})
+	s.Error(err, "Unauthorized request got through")
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestUnauthorizedImageVulnerabilitiesEndpoint() {
+	_, err := s.resolver.ImageVulnerabilities(s.ctx, PaginatedQuery{})
+	s.Error(err, "Unauthorized request got through")
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestUnauthorizedImageVulnerabilityCountEndpoint() {
+	_, err := s.resolver.ImageVulnerabilityCount(s.ctx, RawQuery{})
+	s.Error(err, "Unauthorized request got through")
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestUnauthorizedImageVulnerabilityCounterEndpoint() {
+	_, err := s.resolver.ImageVulnerabilityCounter(s.ctx, RawQuery{})
+	s.Error(err, "Unauthorized request got through")
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestUnauthorizedTopImageVulnerabilityEndpoint() {
+	_, err := s.resolver.TopImageVulnerability(s.ctx, RawQuery{})
+	s.Error(err, "Unauthorized request got through")
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilities() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	vulns, err := s.resolver.ImageVulnerabilities(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(5, len(vulns))
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilitiesFixable() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	query, err := getFixableRawQuery(true)
+	s.NoError(err)
+
+	vulns, err := s.resolver.ImageVulnerabilities(ctx, PaginatedQuery{Query: &query})
+	s.NoError(err)
+	s.Equal(1, len(vulns))
```

Comment:
Expand this and assert `Fixable` and `FixedBy` fields.

Reply:
IsFixable can be checked easily, FixedByVersion is currently querying ComponentCVEEdgeDataStore and requires that the request be scoped by ImageComponent to test, so that can't be done in this test, but could maybe be done in a separate one

Reply:
Okay,  split it out, get the `IsFixable` in and the rest in separate PR.